### PR TITLE
Docs: #601change topic replication factor

### LINKED ISSUE #601: 3Q 22.3: Document changing topic replication factor via API
- https://github.com/redpanda-data/core-internal/issues/18

Document redpanda-data/redpanda/issues/3416. which will greatly simplify increasing/decreasing replication factor on an existing topic via API or RPK

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #601 

preview: 
https://deploy-preview-739--redpanda-documentation.netlify.app/docs/data-management/data-migration/#change-topic-replication-factor
https://deploy-preview-739--redpanda-documentation.netlify.app/docs/deployment/production-deployment/#verify-the-installation
https://deploy-preview-739--redpanda-documentation.netlify.app/docs/cluster-administration/configuration/
https://deploy-preview-739--redpanda-documentation.netlify.app/docs/reference/rpk/rpk-topic/rpk-topic-alter-config/

### CODE REVIEW

File: docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx [hunk 1]
```diff
@@ -7,28 +7,30 @@ title: rpk topic alter-config
...
-2. Deleting a key's value
-3. Appending a new value to a list-of-values key
-4. Subtracting (removing) an existing value from a list-of-values key
-
-The `--dry` option will validate whether the requested configuration change is
-valid, but does not apply it.
+- Setting a key=value pair
+- Deleting a key's value
+- Appending a new value to a list-of-values key
+- Subtracting (removing) an existing value from a list-of-values key
```
Comment (AnneLeeperRP) [inline docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx:17]: Suggestion:
Just say "removing" and leave out "subtracting".

File: docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx [hunk 2]
```diff
@@ -7,28 +7,30 @@ title: rpk topic alter-config
...
 ```
 
-### Flags
+For example, to change the replication factor on an existing topic: 
+
+```
+rpk topic alter-config [TOPICS...] --set replication.factor
+```
+
+When changing the replication factor on existing partitions, the topic remembers the replication factor, so new partitions created in the topic have the new replication factor.
```
Comment (AnneLeeperRP) [inline docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx:31]: How about:
When you change the replication factor on existing partitions, the change applies to new partitions also.

(I don't think a topic can remember anything.)

If the part about the topic is important, maybe lead with "The replication factor is associated with the topic and applies to all its partitions".

File: docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx (hunk 1 above)
Comment (micheleRP) [reply docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx:17]: fixed

File: docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx (hunk 2 above)
Comment (micheleRP) [reply docs/reference/rpk/rpk-topic/rpk-topic-alter-config.mdx:31]: Thx @AnneLeeperRP. I changed this to "The updated replication factor also applies to new partitions created in the topic."

**Comment by micheleRP** [comment]
@mattschumpert @VadimPlh @mmaslankaprv Could you please approve this PR or mark any necessary changes? Thank you!

**Review by VadimPlh — APPROVED** [review]
LGTM (part about replication factor changes)

File: docs/cluster-administration/configuration.mdx [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+title: Cluster Configuration
 ---
 
 <head>
     <meta name="title" content="Cluster configuration | Redpanda Docs"/>
     <meta name="description" content="Cluster configuration general information."/>
 </head>
 
-When you install Redpanda software, configuration properties are automatically set to their default values. These properties are divided into two categories: cluster and node.
+When you install Redpanda software, configuration properties are automatically set to their default values. There are cluster properties, node properties, and topic properties. Some properties can be configured at either the cluster level or the topic level, such as `retention_bytes` (cluster-level) and `retention.bytes` (topic-level). Topic-level properties override cluster-level properties.
```
Comment (Feediver1) [inline docs/cluster-administration/configuration.mdx:10]:
```suggestion
When you install Redpanda software, configuration properties are automatically set to the default values. There are cluster properties, node properties, and topic properties. Some properties can be configured at either the cluster level or the topic level, such as `retention_bytes` (cluster-level) and `retention.bytes` (topic-level). Topic-level properties override cluster-level properties.
```

File: docs/data-management/data-migration.mdx [hunk 4]
```diff
@@ -93,36 +91,44 @@ With this command, MirrorMaker consumes all topics from the redpanda1 cluster an
 MirrorMaker adds the prefix `redpanda1` to the names of replicated topics.
 
 :::note
-MirrorMaker uses the `__consumer_offsets` topic to replicate consumer offsets between clusters. This is currently not supported in Redpanda, but you can follow the progress of this issue at: https://github.com/redpanda-data/redpanda/issues/1752
+- MirrorMaker uses the `__consumer_offsets` topic to replicate consumer offsets between clusters. This is currently not supported in Redpanda, but you can follow the progress of this issue at: https://github.com/redpanda-data/redpanda/issues/1752.
```
Comment (Feediver1) [inline docs/data-management/data-migration.mdx:94]:
```suggestion
- MirrorMaker uses the `__consumer_offsets` topic to replicate consumer offsets between clusters. This is not supported in Redpanda (you can follow the progress of this issue at: https://github.com/redpanda-data/redpanda/issues/1752).
```

File: docs/data-management/data-migration.mdx [hunk 5]
```diff
@@ -93,36 +91,44 @@ With this command, MirrorMaker consumes all topics from the redpanda1 cluster an
...
 :::
 
+## Change topic replication factor
+
+If you need to increase or decrease the replication factor on an existing topic, run:
```
Comment (Feediver1) [inline docs/data-management/data-migration.mdx:99]: This really needs some context--under what circumstances should users increase or decrease the replication factor on an existing topic? For example, why and when would you use a factor greater than 1? Or 2? I know at Confluent the default was 3, but that the replication factor is simply the number of nodes where data is replicated. Worth spelling it out more explicitly here. When I googled this, the most common search was "why is replication factor 3?", which tells me that users don't know the answer:
"In Kafka, replication means that data is written down not just to one broker, but to many. The replication factor is a topic setting and is specified at topic creation time. A replication factor of 3 is a commonly used replication factor as it provides the right balance between broker loss and replication overhead."

File: docs/data-management/data-migration.mdx [hunk 6]
```diff
@@ -137,8 +143,8 @@ Now you know the replication is working!
 
 ## Stop MirrorMaker
 
-To stop the MirrorMaker process, use `top` to find its process ID and then run: `kill <MirrorMaker pid>`
+To stop the MirrorMaker process, use `top` to find its process ID, and then run `kill <MirrorMaker pid>`.
 
 ## Troubleshooting
 
-If you run into any difficulty with data migration, contact us in our [Slack](https://vectorized.io/slack) community so we can help you succeed.
+If you run into any difficulty with data migration, contact us in our [Slack](https://vectorized.io/slack) community.
```
Comment (Feediver1) [inline docs/data-management/data-migration.mdx:150]:
```suggestion
If you run into any difficulty with data migration, you can request help in the Redpanda [Slack](https://vectorized.io/slack) community.
```

File: docs/deployment/production-deployment.mdx [hunk 7]
```diff
@@ -137,6 +137,12 @@ To verify that the cluster is up and running, check the logs:
 journalctl -u redpanda
 ```
 
+If topics were initially created in a test environment with a replication factor of one, use `rpk topic alter-config` to change the topic replication factor:
```
Comment (Feediver1) [inline docs/deployment/production-deployment.mdx:140]:
```suggestion
If topics were initially created in a test environment with a replication factor of `1`, use `rpk topic alter-config` to change the topic replication factor: 
```

**Comment by Feediver1** [comment]
In https://deploy-preview-739--redpanda-documentation.netlify.app/docs/data-management/data-migration/#create-the-mirrormaker-2-config-files please change the heading to use "configuration" instead of "config". Thx

**Review by Feediver1 — APPROVED** [review]
See comments/suggestions

**Review by mmaslankaprv — APPROVED** [review]

File: docs/data-management/data-migration.mdx [hunk 8]
```diff
@@ -93,36 +91,46 @@ With this command, MirrorMaker consumes all topics from the redpanda1 cluster an
 MirrorMaker adds the prefix `redpanda1` to the names of replicated topics.
 
 :::note
-MirrorMaker uses the `__consumer_offsets` topic to replicate consumer offsets between clusters. This is currently not supported in Redpanda, but you can follow the progress of this issue at: https://github.com/redpanda-data/redpanda/issues/1752
+- MirrorMaker uses the `__consumer_offsets` topic to replicate consumer offsets between clusters. This is not supported in Redpanda (you can follow the progress of this issue at: https://github.com/redpanda-data/redpanda/issues/1752).
```
Comment (mattschumpert) [inline docs/data-management/data-migration.mdx:94]: @micheleRP  this is no longer true. We added support for __consumerOffsets in 22.2 , right @mmaslankaprv, @jcsp  ?

File: docs/data-management/data-migration.mdx [hunk 9]
```diff
@@ -137,8 +145,8 @@ Now you know the replication is working!
 
 ## Stop MirrorMaker
 
-To stop the MirrorMaker process, use `top` to find its process ID and then run: `kill <MirrorMaker pid>`
+To stop the MirrorMaker process, use `top` to find its process ID, and then run `kill <MirrorMaker pid>`.
 
 ## Troubleshooting
 
-If you run into any difficulty with data migration, contact us in our [Slack](https://vectorized.io/slack) community so we can help you succeed.
+If you run into any difficulty with data migration, you can request help in the Redpanda [Slack](https://vectorized.io/slack) community.
```
Comment (mattschumpert) [inline docs/data-management/data-migration.mdx:152]: @micheleRP is the slack still vectorized.io?

**Review by mattschumpert — APPROVED** [review]

File: docs/data-management/data-migration.mdx (hunk 8 above)
Comment (micheleRP) [reply docs/data-management/data-migration.mdx:94]: @mattschumpert this got fixed in https://github.com/redpanda-data/documentation/pull/746#event-7726142650. It'll pick it up when these merge

File: docs/data-management/data-migration.mdx (hunk 9 above)
Comment (micheleRP) [reply docs/data-management/data-migration.mdx:152]: Thank you, fixed!

**Pull request merged**